go/pserver/etcd_cclient.go

```diff
@@ -0,0 +1,128 @@
+package pserver
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	DefaultEtcdTimeout time.Duration = 5 * time.Second
+)
+
+type EtcdCClient interface {
```

Comment:
I know it's hard to choose a name, since `EtcdClient` is already used ;)
However, `EtcdCClient` may not be a good name, because C Client means it's for C. But here is not really related to C (Go can use it as well). Maybe `ClientEtcd`?